hal/halx86/smp/i386/apentry.S

```diff
@@ -0,0 +1,73 @@
+#include <asm.inc>
+
+PUBLIC _APEntry
+PUBLIC _APEntryEnd
+
+.code16
+_APEntry:
+    cli
+
+    xor     ax, ax
+	mov		ds, ax
+	mov		ss, ax
+    mov     fs, ax
+    mov     gs, ax
+
+    /* Load the GDT and blank IDT */
+#ifdef _USE_ML
+    lgdt fword ptr cs:[gdtptr - _APEntry]
+    lidt fword ptr cs:[idtdesc - _APEntry]
+#else
+    lgdt cs:[gdtptr - _APEntry]
+    lidt cs:[idtdesc - _APEntry]
+#endif
+
+    mov   eax, cr0
+    or    eax, HEX(00000001)
+    mov   cr0, eax
+
+    ljmp16 HEX(08), (HEX(4000) + _APEntryEnd - _APEntry)
+
+    hlt
+
+gdtptr:
```

Comment:
Why is this using it's own GDT? Why doesn't it reuse the one from the BSP?

Reply:
It does eventually reuse the one from the BSP, but when you are initializing an application processor you need to create a temporary GDT that can be accessed in the real mode stub.

Reply:
Nevermind, in fact each CPU needs it's own GDT (for FS and TSS)